### ISSUE TITLE
[Bug]: Metadata menu for product type - 404

### ISSUE THREAD

**Issue by littletuna4** [medusajs/medusa#10762]
### Package.json file

```JSON
{
  "name": "medusa-starter-default",
  "version": "0.0.1",
  "description": "A starter for Medusa projects.",
  "author": "Medusa (https://medusajs.com)",
  "license": "MIT",
  "keywords": [
    "sqlite",
    "postgres",
    "typescript",
    "ecommerce",
    "headless",
    "medusa"
  ],
  "scripts": {
    "build": "medusa build && ln -s .medusa/server/public/ public",
    "seed": "medusa exec ./src/scripts/seed.ts",
    "start": "medusa start",
    "dev": "medusa develop",
    "predeploy": "medusa db:migrate",
    "test:integration:http": "TEST_TYPE=integration:http NODE_OPTIONS=--experimental-vm-modules jest --silent=false --runInBand --forceExit",
    "test:integration:modules": "TEST_TYPE=integration:modules NODE_OPTIONS=--experimental-vm-modules jest --silent --runInBand --forceExit",
    "test:unit": "TEST_TYPE=unit NODE_OPTIONS=--experimental-vm-modules jest --silent --runInBand --forceExit"
  },
  "dependencies": {
    "@aws-sdk/client-ses": "^3.716.0",
    "@medusajs/admin-sdk": "^2.0.0",
    "@medusajs/cli": "^2.0.0",
    "@medusajs/framework": "^2.0.0",
    "@medusajs/medusa": "^2.1.3",
    "@medusajs/payment-stripe": "^2.0.4",
    "@mikro-orm/core": "5.9.7",
    "@mikro-orm/knex": "5.9.7",
    "@mikro-orm/migrations": "5.9.7",
    "@mikro-orm/postgresql": "5.9.7",
    "@react-email/components": "0.0.31",
    "@tanstack/react-query": "^5.59.0",
    "awilix": "^8.0.1",
    "fast-uri": "^3.0.3",
    "lodash": "^4.17.21",
    "nodemailer": "^6.9.16",
    "pg": "^8.13.0",
    "react-dropzone": "^14.2.3",
    "zod": "3.22.4"
  },
  "devDependencies": {
    "@medusajs/test-utils": "^2.0.0",
    "@mikro-orm/cli": "5.9.7",
    "@swc/core": "1.5.7",
    "@swc/jest": "^0.2.36",
    "@tanstack/eslint-plugin-query": "^5.58.1",
    "@types/jest": "^29.5.13",
    "@types/lodash": "^4",
    "@types/node": "^20.0.0",
    "@types/nodemailer": "^6",
    "@types/react": "^18.3.2",
    "@types/react-dom": "^18.2.25",
    "eslint": "^9.12.0",
    "jest": "^29.7.0",
    "prop-types": "^15.8.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-email": "3.0.4",
    "ts-node": "^10.9.2",
    "typescript": "^5.6.2",
    "vite": "^5.2.11"
  },
  "engines": {
    "node": ">=20"
  },
  "packageManager": "yarn@4.3.1"
}
```

### Node.js version

v22.11.0

### Database and its version

Neon (PostgreSQL 17)

### Operating system name and version

WSL 2

### Browser name

Edge

### What happended?

TLDR: The same as #10584 except product type instead of product variant.

The metadata page for a product type returned a 404 error

Unable to access the metadata or metadata/edit pages for a product variant with a 404. 

### Expected behavior

Menu to edit product type metadata

### Actual behavior

404 Page

### Link to reproduction repo

-

**Comment by olivermrbl** [comment]
@littletuna4, can I get you to upgrade all your Medusa packages to the latest version and try again?

I can't reproduce the issue, so closing this for now. Feel free to reopen if the issue persists.

**Comment by littletuna4** [comment]
Hmmm - thanks for having a look,

Issue persists with upgraded deps:
```
...
    "@medusajs/admin-sdk": "^2.1.3",
    "@medusajs/cli": "^2.1.3",
    "@medusajs/framework": "^2.1.3",
    "@medusajs/medusa": "^2.1.3",
    "@medusajs/payment-stripe": "^2.1.3",
...
```
and with a fresh .medusa folder

Page that can't be accessed (404):
[http://localhost:9000/app/settings/product-types/ptyp_01JFYVQKVS0GY01BPJHHR7XQM8/metadata/edit](http://localhost:9000/app/settings/product-types/ptyp_01JFYVQKVS0GY01BPJHHR7XQM8/metadata/edit)

Could it be to do with using the custom module in the fashion starter (or is that a stretch)?

**Comment by MarcosRava** [comment]
I Have the same issue
